operator/cmd/mesh/uninstall.go

```diff
@@ -0,0 +1,182 @@
+	l := clog.NewConsoleLogger(cmd.OutOrStdout(), cmd.ErrOrStderr(), installerScope)
+	if err := configLogs(logOpts); err != nil {
+		return fmt.Errorf("could not configure logs: %s", err)
+	}
+	restConfig, _, clt, err := K8sConfig(uiArgs.kubeConfigPath, uiArgs.context)
+	if err != nil {
+		return err
+	}
+
+	if err := uninstallWrapper(cmd, rootArgs, uiArgs, restConfig, clt, l); err != nil {
+		return fmt.Errorf("failed to uninstall control plane: %v", err)
+	}
+	return nil
+}
+
+// uninstallWrapper is helper function to uninstalls control plane by either
+// pruning by target revision or deleting specified manifests.
+func uninstallWrapper(cmd *cobra.Command, rootArgs *rootArgs, uiArgs *uninstallArgs,
+	restConfig *rest.Config, clt client.Client, l clog.Logger) error {
+	cache.FlushObjectCaches()
+	opts := &helmreconciler.Options{DryRun: rootArgs.dryRun, Log: l, ProgressLog: progress.NewLog()}
+	h, err := helmreconciler.NewHelmReconciler(clt, restConfig, nil, opts)
+	if err != nil {
+		return fmt.Errorf("failed to create reconciler: %v", err)
+	}
+
+	if uiArgs.revision != "" {
+		objectsList, tp, err := h.GetPrunedResourcesList(uiArgs.revision)
+		if err != nil {
+			return err
+		}
+		if err := preCheckWarnings(cmd, rootArgs, uiArgs, uiArgs.revision, tp); err != nil {
+			return fmt.Errorf("failed to do preuninstall check: %v", err)
+		}
+
+		if err := h.PruneControlPlaneByRevision(uiArgs.revision, objectsList); err != nil {
+			return fmt.Errorf("failed to prune control plane resources by revision: %v", err)
+		}
+	} else if uiArgs.filename != "" {
+		manifestMap, iops, err := manifest.GenManifests([]string{uiArgs.filename}, applyFlagAliases(uiArgs.set, uiArgs.manifestsPath, uiArgs.revision), uiArgs.force, restConfig, l)
+		if err != nil {
+			return err
+		}
+		if iops == nil {
+			return fmt.Errorf("istioOperatorSpec is nil")
+		}
+		if err := preCheckWarnings(cmd, rootArgs, uiArgs, iops.Revision, nil); err != nil {
+			return fmt.Errorf("failed to do preuninstall check: %v", err)
+		}
+		cpManifests := manifestMap[name.PilotComponentName]
+		if err := h.DeleteControlPlaneByManifests(strings.Join(cpManifests, "---"), iops.Revision); err != nil {
+			return fmt.Errorf("failed to delete control plane by manifests: %v", err)
+		}
+	}
+	opts.ProgressLog.SetState(progress.StateComplete)
+	return nil
+}
+
+// preCheckWarnings checks possible breaking changes and issue warnings to users, it checks the following:
+// 1. checks proxies still pointing to the target control plane revision.
+// 2. lists to be pruned resources if user uninstall by --revision flag.
+func preCheckWarnings(cmd *cobra.Command, rootArgs *rootArgs, uiArgs *uninstallArgs, rev string, resourcesList []string) error {
+	pids, err := proxyinfo.GetIDsFromProxyInfo(uiArgs.kubeConfigPath, uiArgs.context, rev, uiArgs.istioNamespace)
+	if err != nil {
+		return err
+	}
+	if uiArgs.skipConfirmation || uiArgs.force || rootArgs.dryRun {
```

Comment:
dry-run would do everything _except_ actually delete the objects. the usual approach is just to output something like "Deleting foo skipped because of dry-run".
skipConfirmation would just automatically give a "Y" every time that check function is called.